File: src/WebClient/src/app/core/models/playlist/playlistResponse.ts
import { ArtistResponse } from "../user/artistResponse";
import { SongResponse } from "../song/songResponse";


export interface PlaylistResponse {
  playlistId: number;
  name: string;
  description?: string;
  isPrivate: boolean;
  totalDuration: number;
  songsCount: number;
  likesCount: number;
  createdAt: string;
  songs: PlaylistSongResponse[];
  photoUrl?: string;
  user: ArtistResponse;
}

export interface PlaylistSongResponse {
  position: number;
  song: SongResponse;
}

File: src/WebApp/Application/Models/Playlists/PlaylistResponse.cs
using Application.Models.Users;

namespace Application.Models.Playlists;
public sealed class PlaylistResponse
{
    public int PlaylistId { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public bool IsPrivate { get; set; } = false;
    public int TotalDuration { get; set; }
    public int SongsCount { get; set; } = default;
    public int LikesCount { get; set; } = default;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public List<PlaylistSongResponse> Songs { get; set; } = [];
    public string? PhotoUrl { get; set; }
    public ArtistResponse User { get; set; } = null!;
}
